private/pkg/slogtest/slogtest.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package slogtest provides Loggers for testing.
+package slogtest
+
+import (
+	"log/slog"
+	"os"
+	"testing"
+
+	"github.com/bufbuild/buf/private/pkg/slogbuild"
+	"github.com/stretchr/testify/require"
+)
+
+// NewLogger returns a new Logger for testing.
+func NewLogger(t testing.TB) *slog.Logger {
```

Comment:
Wdyt about adding functional options to tune logging in testing, specifically log level is one we use often.

https://pkg.go.dev/go.uber.org/zap/zaptest#LoggerOption

Reply:
Sure, give me a second

Reply:
Done